pkg/drivers/qemu/qemu.go

```diff
@@ -712,6 +738,26 @@ func (d *Driver) generateUserdataDisk(userdataFile string) (string, error) {
 	return ccRoot, nil
 }
 
+// generateExtraDisk creates the extra disk files
+func (d *Driver) generateExtraDisk(i int) error {
+	diskPath := d.extraDiskPath(i)
+	log.Infof("Creating an extra %d MB raw hard disk image...", d.DiskSize)
+
+	if _, err := os.Stat(diskPath); os.IsNotExist(err) {
+		file, err := os.OpenFile(diskPath, os.O_CREATE|os.O_EXCL|os.O_WRONLY, 0644)
+		if err != nil {
+			return errors.Wrap(err, "open")
+		}
+		defer file.Close()
+
+		if err := file.Truncate(util.ConvertMBToBytes(d.DiskSize)); err != nil {
+			return errors.Wrap(err, "truncate")
+		}
```

Comment:
This works but running "qemu-img create -f raw name size" is simpler and
does the right thing. For example is always allocates the first 4k bytes
to allow detection of logical block size.

Reply:
It seems to me that this comment and the one about making a generic helper for creating a raw disk ([here](https://github.com/kubernetes/minikube/pull/15887#discussion_r1112285355)) are mutually exclusive. I'd prefer to create a helper that doesn't rely on qemu tools. Ideally, generating all raw disks the same way will mean there is no behavior discrepancy between drivers.